modules/apps/client-extension/client-extension-web/src/main/java/com/liferay/client/extension/web/internal/display/context/ViewClientExtensionEntryPartDisplayContext.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.client.extension.web.internal.display.context;
+
+import com.liferay.client.extension.type.CET;
+import com.liferay.client.extension.type.annotation.CETProperty;
+import com.liferay.portal.kernel.language.LanguageUtil;
+import com.liferay.portal.kernel.util.CamelCaseUtil;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.lang.reflect.Method;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import javax.portlet.PortletRequest;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class ViewClientExtensionEntryPartDisplayContext<T extends CET> {
+
+	public ViewClientExtensionEntryPartDisplayContext(
+		T cet, PortletRequest portletRequest) {
+
+		_cet = cet;
+		_portletRequest = portletRequest;
+	}
+
+	public T getCET() {
+		return _cet;
+	}
+
+	public String getLabel(Method method) {
+		CETProperty cetProperty = method.getAnnotation(CETProperty.class);
+
+		String label = cetProperty.label();
+
+		if (Validator.isBlank(label)) {
+			label = CamelCaseUtil.fromCamelCase(cetProperty.name());
+		}
+
+		return LanguageUtil.get(_getHttpServletRequest(), label);
+	}
+
+	public Collection<Method> getPropertyMethods() {
+		Class<? extends CET> clazz = _cet.getClass();
+
+		List<Method> methods = new ArrayList<>();
+
+		for (Class<?> iface : clazz.getInterfaces()) {
+			if ((iface == _CET_CLASS) || !_CET_CLASS.isAssignableFrom(iface)) {
```

Comment:
I'm not 100% sure if `isAssignableFrom` works in case `iface` is provided by other osgi  bundle, different from `client-extension-type-api`. Maybe we won't face that case, but if we plan others to add new CETs in separate bundles, we need to make sure

Reply:
> Maybe we won't face that case

Not really :sweat_smile: -> https://github.com/liferay/liferay-portal/blob/9514a095c668ebd7e76145e149a9d19bd80f01de/modules/apps/client-extension/client-extension-type-impl/src/main/java/com/liferay/client/extension/type/internal/factory/CETFactoryImpl.java#L55-L83

> if we plan others to add new CETs in separate bundles

I'd say we won't, since this UI has narrowed its scope a lot. In theory it will only be used for infrastructure extensions because the other types will be configured in a different place. I even think that there are already client extensions out there that don't use this framework completely, just the DB schema (but would need to check, I'm not 100% sure).

So, unless we decide to revamp this in the future, I'm OK with how it is now :shrug: